### ISSUE TITLE
ALSA and JACK do not support large SysEx messages

### ISSUE THREAD

**Issue by keinstein** [thestk/rtmidi#214]
As there were some patches that improve the support of larger SysEx messages on different backends  (e.g. #91 ). However neither ALSA nor JACK support these kind of messages. While the ALSA sequencer theoretically allows larger SysEx messages. The kernel silently clears the input buffer as soon as an event could not be delivered to it. This means when the sender is to fast by a single packet about 2-3k of SysEx data gets silently dropped, when communicating with a software device.

A workaround could be to implement RawMIDI support for ALSA.
Another workaround could be to implement user defined input/output pool sizes.

**Comment by KoSv** [comment]
hi there, 
I'm trying to send around 6k and my hardware device freezes (sysex massage is not complete)
where receiving is not a problem!
Is there a workaround?
greets and happy christmas! :)

**Comment by jcelerier** [comment]
Hello,
I have a partial work on a raw midi backend with support for chunking on the https://github.com/jcelerier/libremidi fork, testing would be appreciated.

**Comment by insolace** [comment]
Hello,

I've confirmed that the libremidi fork works. The RAW send method can send large (60-100k) sysex payloads (firmware), the current RtMidi ALSA send method cannot handle more than 1-2k messages.

**Comment by jcelerier** [comment]
yay :)

**Comment by insolace** [comment]
> yay :) 

@jcelerier Any chance you'd be willing to port your fix into rtmidi and make a pull request?

**Comment by jcelerier** [comment]
Well, it's more of a complete rewrite than a fix, I just don't have enough time for that sorry :(

**Comment by insolace** [comment]
I figured as much when I looked at your Raw Send code. I wish I had the time to do this myself, but for now we will have to hold off on releasing Linux versions of our editors until this issue is fixed.

**Comment by keinstein** [comment]
@jcelerier Did I read libremidi right that you have the possibility to configure the transfer rate by inserting an artificial sleep in the ALSA sending routine? This could be a compromise which is much easier to backport.

**Comment by jcelerier** [comment]
yes pretty much, but that's with the "raw" alsa api, not the seq one (I remember trying and failing to make it work there though I could just have missed something simple... that was some months ago, it's not entirely clear in my mind). All the relevant code is there, feel free to backport: https://github.com/jcelerier/libremidi/blob/master/include/libremidi/detail/raw_alsa.hpp#L447
